### DISCUSSION TITLE
Move or export the image

### DISCUSSION THREAD

**Discussion by ai30328348** [photoprism/photoprism#4302] [Ideas]
I don't want to see a picture file saved by date in a folder because my picture is already in a folder with the correct meaning, and the picture name is named with the correct meaning.
If I use the import function, my images will all be mixed together and will lose their original meaning. If you don't use the import feature, the newly added image files won't be stored in their original meaning.
To do this, I'm requesting to add the ability to reorganize image files, such as file moves, or to allow the import feature to define storage rules, such as folders mapping to albums, and recognize folder names as labels. In addition, the uncategorized pictures of the album are not refreshed in time after the operation.
If that's difficult,Expect to add an export feature to export folders with custom rules and recover the names of picture files such as album_1/web.jpg, tag_1/word.jpg.

**Comment by lastzero** [comment]
If your pictures are already properly named, sorted, and in the right folders, why do you use the (optinal) import functionality in the first place?

**Reply by ai30328348** [reply]
If you don't save the picture according to the fixed rules, it will gradually become more and more confusing。So the import function is preferred
When adding new pictures, they may be placed in the wrong folder and I need to correct them. I want to put the pictures in the right folder, I need to manually move the pictures with other software, and then reindex the pictures

**Reply by lastzero** [reply]
You can move them however you like. By adding, do you mean uploading new files using the web upload dialog?

**Reply by ai30328348** [reply]
I synced the pictures to the specified folders through nextcloud and then scanned them with photosprim, the folders weren't the right ones and I needed to move them to the right folders

**Comment by ai30328348** [comment]
I need to put the pictures in the specified folder. I want to move the original image directly through the prism, I don't want to use other software to manually move the image and reindex the image. If you only use the indexing function, you will forget where the image is, so you need to use the import function.
Actually, I'm not against encrypting file names and storing encrypted folders. But I want my picture files to be restored to the storage method of the unencrypted name at any time, and to store them according to the custom rules. 
**I think the way images are stored should be more open and versatile.**

**Comment by graciousgrey** [comment]
> I want to move the original image directly through the prism, I don't want to use other software to manually move the image and reindex the image.

You don't need to install extra software to move your files. You can do so with the Finder on Mac or the Explorer on Windows. 

https://docs.photoprism.app/user-guide/sync/webdav/#__tabbed_1_1

Providing a UI for this might be possible in the future but this is nothing we can work on any time soon due to all the other tasks on our to do.

>Actually, I'm not against encrypting file names and storing encrypted folders. But I want my picture files to be restored to the storage method of the unencrypted name at any time, and to store them according to the custom rules.

I assume with encryption you are referring to the unique names created on import. If you don't want PhotoPrism to change the names or the folder structure of your files I would recommend using the index method instead of the import method.

https://docs.photoprism.app/user-guide/library/

Making the import path/naming scheme configurable is already planned:

- https://github.com/photoprism/photoprism/issues/475

The original name of each file is stored in the database and also shown in the edit dialog within the web ui. So you might be able to write a script or some code to rename your files according to this. 

However renaming imported files back to the original name would come with a lot of risks and potentially needed conflict resolutions as it might be possible that multiple files with the same original name are located in the same directory after import so it would not be possible to rename both back as one would need to be overwritten. This is one of the reasons why imported files get a unique name.

**Reply by ai30328348** [reply]
If I use indexing in photoprism I find that the image is not in the right folder, I need to remember the file name and use the file manager to move it, if there are too many images it is very difficult, it is not as convenient as using the file manager directly, if just this photoprism does not seem much use to me
If you only use indexes, the storage path of images will be cluttered and not conducive to management. So sooner or later you need to use the import function, I hope that there is an import function and an export function, and the export rules can be customized. Maybe it's similar to the download function, but it's done on the server, or it's just a custom rule.
I don't want the original picture file to be used only in photoprism

**Reply by ai30328348** [reply]
As for the possibility of the same file in a folder, just add a timestamp to the new file name.
Maybe you should consider changing the import function first, it doesn't recognize the folder as an album or tag, I have to manually add it to the album or tag again.
Maybe it's better to use a script to implement a custom export feature. But hopefully it can be implemented in a light prism.
Thank you for your answers
